test/unit/spec/LoginRouter_spec.js

```diff
@@ -539,6 +542,31 @@ function (Okta, Q, Logger, OktaAuth, Util, Expect, Router,
         expect(test.router.navigate).toHaveBeenCalledWith('', { trigger: true });
       });
     });
+    itp('triggers a "navigated" event when routing to default route and when status is UNAUTHENTICATED', function () {
+      return setup({ stateToken: 'aStateToken' })
+      .then(function (test) {
+        Util.mockRouterNavigate(test.router);
+        test.setNextResponse(resUnauthenticated);
+        test.router.navigate('/app/sso');
+        return Expect.waitForPrimaryAuth(test);
+      })
+      .then(function (test) {
+        expect(test.router.navigate).toHaveBeenCalledWith('', { trigger: true });
+        expect(test.navigatedSpy.calls.count()).toBe(2);
+        expect(test.navigatedSpy.calls.allArgs()[0]).toEqual([{
+          view: '/context.html',
+          controller: 'refresh-auth-state',
+          transaction: undefined
+        }]);
+        expect(test.navigatedSpy.calls.allArgs()[1]).toEqual([{
+          view: '/context.html',
+          controller: 'primary-auth',
+          transaction: jasmine.objectContaining({
```

Comment:
is there chance to do exact match or is it because too much noise coming from the Transaction class

Reply:
Pushed an update to assert the entire object. Good reminder that we should check for as many values as possible.

Reply:
also I'm wondering if we expose too much information by having
- `authenticate` function (probably others per different transaction
- `data` the raw response

was the intention is expose as much information as possible and let user do whatever they want?
my worry is too flexible unless it's a clear request from users (I may miss some information)